### PR TITLE
Add Python Tests

### PR DESCRIPTION
**Pull request by igor-egorov**
### Description of the Change

Add new tests for Iroha Python bindings.

### Benefits

New test set cover building of Iroha's queries and commands. Each parameter of either command or query is tested separately from other parameters for boundary values (in a moment only one method's parameter takes invalid value - the rest are provided with valid values).

The total number of tests for commands is 98, queries - 45.

| Iroha Command | Tests |
| --- | --- |
| | test_empty_tx |
| **AddPeer** | test_add_peer<br/> test_add_peer_valid_domains<br/> test_outdated_add_peer<br/> test_add_peer_with_invalid_creator<br/> test_add_peer_with_invalid_creator_domain<br/> test_add_peer_with_empty_creator<br/> test_add_peer_with_invalid_key_size<br/> test_add_peer_with_invalid_domain |
| **AddSignatory** | test_add_signatory<br/> test_add_signatory_invalid_account_name<br/> test_add_signatory_empty_account<br/> test_add_signatory_invalid_domain<br/> test_add_signatory_invalid_key |
| **AddAssetQuantity** | test_add_asset_quantity<br/> test_add_asset_quantity_valid_account_and_asset<br/> test_add_asset_quantity_invalid_account<br/> test_add_asset_quantity_invalid_account_domain<br/> test_add_asset_quantity_empty_account<br/> test_add_asset_quantity_invalid_asset<br/> test_add_asset_quantity_invalid_asset_domain<br/> test_add_asset_quantity_empty_asset<br/> test_add_asset_quantity_invalid_amount |
| **RemoveSignatory** | test_remove_signatory<br/> test_remove_signatory_empty_account<br/> test_remove_signatory_invalid_account<br/> test_remove_signatory_invalid_account_domain<br/> test_remove_signatory_invalid_key |
| **CreateAccount** | test_create_account<br/> test_create_account_invalid_account<br/> test_create_account_invalid_domain<br/> test_create_account_invalid_key |
| **CreateDomain** | test_create_domain<br/> test_create_domain_with_invalid_name<br/> test_create_domain_invalid_role |
| **SetAccountQuorum** | test_set_account_quorum<br/> test_set_account_quorum_invalid_account<br/> test_set_account_quorum_invalid_account_domain<br/> test_set_account_quorum_empty_account<br/> test_set_account_quorum_invalid_quantity |
| **TransferAsset** | test_transfer_asset<br/> test_transfer_asset_with_invalid_name<br/> test_transfer_asset_with_invalid_domain<br/> test_transfer_asset_with_empty_name<br/> test_transfer_asset_invalid_from_account<br/> test_transfer_asset_invalid_from_account_domain<br/> test_transfer_asset_empty_from_account<br/> test_transfer_asset_invalid_to_account<br/> test_transfer_asset_invalid_to_account_domain<br/> test_transfer_asset_empty_to_account<br/> test_transfer_asset_description_valid_values<br/> test_transfer_asset_invalid_description<br/> test_transfer_asset_maximum_amount |
| **SetAccountDetail** | test_set_account_detail<br/> test_set_account_detail_invalid_account<br/> test_set_account_detail_invalid_account_domain<br/> test_set_account_detail_empty_account<br/> test_set_account_detail_valid_key<br/> test_set_account_detail_invalid_key<br/> test_set_account_detail_valid_value<br/> test_set_account_detail_oversized_value |
| **AppendRole** | test_append_role<br/> test_append_role_invalid_account<br/> test_append_role_invalid_domain<br/> test_append_role_empty_account<br/> test_append_role_with_invalid_name |
| **CreateAsset** | test_create_asset<br/> test_create_asset_invalid_name<br/> test_create_asset_invalid_domain<br/> test_create_asset_zero_precision |
| **CreateRole** | test_create_role<br/> test_create_role_with_invalid_name<br/> test_create_role_with_empty_permissions<br/> test_create_role_wrong_permissions<br/> test_create_role_wrong_permissions_2 |
| **DetachRole** | test_detach_role<br/> test_detach_role_with_invalid_account<br/> test_detach_role_with_invalid_account_domain<br/> test_detach_role_with_empty_account<br/> test_detach_role_with_invalid_name |
| **GrantPermission** | test_grant_permission<br/> test_grant_permission_empty_account<br/> test_grant_permission_invalid_account<br/> test_grant_permission_invalid_account_domain<br/> test_grant_permission_with_invalid_name |
| **RevokePermission** | test_revoke_permission<br/> test_revoke_permission_invalid_account<br/> test_revoke_permission_invalid_account_domain<br/> test_revoke_permission_empty_account<br/> test_revoke_permission_with_invalid_name |
| **SubtractAssetQuantity** | test_subtract_asset_quantity<br/> test_subtract_asset_quantity_invalid_account<br/> test_subtract_asset_quantity_invalid_account_domain<br/> test_subtract_asset_quantity_with_empty_account<br/> test_subtract_asset_quantity_invalid_asset_name<br/> test_subtract_asset_quantity_invalid_asset_domain<br/> test_subtract_asset_quantity_empty_account<br/> test_subtract_asset_quantity_invalid_amount |

| Iroha Query | Tests |
| --- | --- |
| | test_empty_query |
| **GetAccount** | test_get_account<br/> test_outdated_get_account<br/> test_get_account_with_invalid_creator<br/> test_get_account_with_invalid_creator_domain<br/> test_get_account_with_empty_creator_domain<br/> test_get_account_invalid_name<br/> test_get_account_invalid_domain<br/> test_get_account_with_empty_name |
| **GetSignatories** | test_get_signatories<br/> test_get_signatories_invalid_account<br/> test_get_signatories_invalid_domain<br/> test_get_signatories_empty_account |
| **GetAccountTransactions** | test_get_account_transactions<br/> test_get_account_transactions_invalid_name<br/> test_get_account_transactions_invalid_domain<br/> test_get_account_transactions_with_empty_account |
| **GetAccountAssetTransactions** | test_get_account_asset_transactions<br/> test_get_account_asset_transactions_invalid_account<br/> test_get_account_asset_transactions_invalid_account_domain<br/> test_get_account_asset_transactions_invalid_asset_name<br/> test_get_account_asset_transactions_invalid_asset_domain<br/> test_get_account_asset_transactions_empty_asset |
| **GetAccountAssets** | test_get_account_assets<br/> test_get_account_assets_invalid_account<br/> test_get_account_assets_invalid_account_domain<br/> test_get_account_assets_invalid_asset_name<br/> test_get_account_assets_invalid_asset_domain<br/> test_get_account_assets_empty_asset |
| **GetRoles** | test_get_roles |
| **GetAssetInfo** | test_get_asset_info<br/> test_get_asset_info_invalid_name<br/> test_get_asset_info_invalid_domain<br/> test_get_asset_info_empty_asset_name |
| **GetRolePermissions** | test_get_role_permissions<br/> test_get_role_permissions_with_invalid_name |
| **GetTransactions** | test_get_transactions<br/> test_get_transactions_with_empty_vector<br/> test_get_transactions_with_invalid_hash_sizes<br/> test_get_transactions_with_one_valid_and_one_invalid_hash_1<br/> test_get_transactions_with_one_valid_and_one_invalid_hash_2 |
| **GetAccountDetail** | test_get_account_detail<br/> test_get_account_detail_invalid_name<br/> test_get_account_detail_invalid_domain<br/> test_get_account_detail_with_empty_name |

### Possible Drawbacks 

Summary tests' compilation and execution time is about 170 seconds.

### Usage Examples or Tests

Before all, please ensure that `google.protobuf` package is installed for Python 3.
You can do this using the command: `python3 -c "import google.protobuf"` 
If zero output was produced, then package is installed. If there are errors, please install the package using the following commands:

```
python3 -m pip install google --user
python3 -m pip install protobuf --user
```

#### Testing

```
cmake -H. -Bbuild -DSWIG_PYTHON=ON
cmake --build build --target irohapy -- -j4
cmake --build build --target python_tests -- -j4
cd build
ctest -R python --output-on-failure
```

### CODE REVIEW

File: test/module/shared_model/bindings/query-test.py [hunk 1]
```diff
@@ -46,51 +122,291 @@ def check_proto_query(self, blob):
...
+    for name in INVALID_NAMES_1:
+      with self.assertRaises(ValueError):
+        self.base().getAccountAssetTransactions("{}@test".format(name), "coin#test").build()
+
+  def test_get_account_asset_transactions_invalid_account_domain(self):
+    for domain in INVALID_DOMAINS:
+      with self.assertRaises(ValueError):
+        self.base().getAccountAssetTransactions("admin@{}".format(domain), "coin#test").build()
+
+  def test_get_account_asset_transactions_empty_aссount(self):
```
Comment (lebdron) [inline test/module/shared_model/bindings/query-test.py]: `aссount` -> `account` (first word contains Cyrillic letters)

File: test/module/shared_model/bindings/query-test.py [hunk 2]
```diff
@@ -46,51 +122,291 @@ def check_proto_query(self, blob):
...
+    for name in INVALID_NAMES_1:
+      with self.assertRaises(ValueError):
+        self.base().getAccountAssets("{}@test".format(name), "coin#test").build()
+
+  def test_get_account_assets_invalid_account_domain(self):
+    for domain in INVALID_DOMAINS:
+      with self.assertRaises(ValueError):
+        self.base().getAccountAssets("admin@{}".format(domain), "coin#test").build()
+
+  def test_get_account_assets_empty_aссount(self):
```
Comment (lebdron) [inline test/module/shared_model/bindings/query-test.py]: `aссount` -> `account` (first word contains Cyrillic letters)
